src/app-router.js

```diff
@@ -240,26 +240,28 @@
   function importAndActivate(router, importUri, route, url, eventDetail) {
     var importLink;
     function importLoadedCallback() {
+      route.importInProgress = false;
+      importedURIs[importUri]  = 'loaded';
       activateImport(router, importLink, importUri, route, url, eventDetail);
     }
 
     if (!importedURIs.hasOwnProperty(importUri)) {
       // hasn't been imported yet
-      importedURIs[importUri] = true;
+      importedURIs[importUri] = 'loading';
```

Comment:
Could we leave this as a boolean and add a second boolean just below here `importLink.loading = false`?